### ISSUE TITLE
Event for clearable

### ISSUE THREAD

**Issue by itlpps** [sagalbot/vue-select#1429]
I want get an event when click on "x" for clear

**Comment by darrinmn9** [comment]
+1

**Comment by vitornovo54** [comment]
+1

**Comment by LucasVanni** [comment]
+1

**Comment by itlpps** [comment]
I solved this with [patch-package](https://www.npmjs.com/package/patch-package).

Replace node_module/vue-select/dist/vue-select.js for this [content vue-select.js](https://gist.github.com/itlpps/31bbb418bc164a0eafe98c19dcb0d9f8) and apply a new patch.

**Comment by vitornovo54** [comment]
Or you can resolve this in a more elegant way. 
Vue-selects offer a way to define the component. So you can simple use vue render function to mount any event. I currently creating a render function then adding an $emit. Since it's a component, all you need to do is to listen it in the parent. It's more work but it will maintain the package without actually patching it.

in your view
`<v-select :components="{Deselect}" />`

in your vue or component. I'm currently using in a component. 
` 
...
data() { 
    return {
      Deselect: {
        render: (createElement) => {
          return createElement('span', {
            on: {
              click: ()=>{
                this.$emit('deselect',this);
              }
            }
          },'❌');
        }
      }
}
...`

Then you can mount a listener for example in the created state.
`...
created(){
      this.$on('deselect',this.someMethod);
},
...`
[https://vue-select.org/guide/components.html#deselect](https://vue-select.org/guide/components.html#deselect)

**Comment by darrinmn9** [comment]
I realize there are possible work arounds, so thanks to those who commented. but it just strikes me as odd that a component exposes an "X" button that you can click, wouldn't expose a custom event to listen for that button click.

**Comment by vitornovo54** [comment]
yeah, me too 😩. That's why I'm doing this way. It will be updated once available without yet including another package. As you just saw, you're not alone in this topic 😆

**Comment by danimalweb** [comment]
I also need this :)

**Comment by SaqibSyed1014** [comment]
> I solved this with [patch-package](https://www.npmjs.com/package/patch-package).
> 
> Replace node_module/vue-select/dist/vue-select.js for this [content vue-select.js](https://gist.github.com/itlpps/31bbb418bc164a0eafe98c19dcb0d9f8) and apply a new patch.
> 
> Usage:
> 
> ```js
> <v-select
>     ...
>     @option:clear="onClear"
> />
> ```

Is this fix added to the current package yet?

**Comment by ecuci** [comment]
+1

**Comment by camille500** [comment]
+1

**Comment by itlpps** [comment]
One year later, using the same component in another project and still don't have this event :(
I made a fork and implemented this for myself, **using vue 3 (v4.0.0-beta.3)**

`yarn add git+https://github.com/itlpps/vue-select.git#event-cleared`

usage:
```
<v-select
    ....
    @option:cleared="onClear"
></v-select>
```

**Comment by ImamAlfariziSyahputra** [comment]
> One year later, using the same component in another project and still don't have this event :( I made a fork and implemented this for myself, **using vue 3 (v4.0.0-beta.3)**
> 
> `yarn add git+https://github.com/itlpps/vue-select.git#event-cleared`
> 
> usage:
> 
> ```
> <v-select
>     ....
>     @option:cleared="onClear"
> ></v-select>
> ```

thanks @itlpps it works, but console gave me a warning like this : "option:cleared" but it is neither declared in the emits option nor as an "onOption:cleared" prop.

How to fix this by the way?

**Comment by igorespin** [comment]
+1

**Comment by sharan-mohandas-tfs** [comment]
Since we don't have an explicit event yet, pasting the snippet of the workaround I used.

```
watch: {
    search(value) {
        if (value === null) {
            // Your logic here
        }
    },
}
```

`<v-select v-model="search">`

**Comment by gotoworld** [comment]
@clear="handleQuery"

**Comment by Iterator98** [comment]
Any update here?

**Comment by Ali7med** [comment]
Any update here?

**Comment by birdhouses** [comment]
Any update here?

**Comment by Ali7med** [comment]
Any update here?

**Comment by baraaksayeth** [comment]
Is there any solution?
